services/azure-devops/azure-devops-artifacts.service.js

```diff
@@ -0,0 +1,102 @@
+'use strict'
+
+const Joi = require('joi')
+const { version: versionColor } = require('../color-formatters')
+const { addv } = require('../text-formatters')
+const {
+  renderVersionBadge: renderNugetVersionBadge,
+} = require('../nuget/nuget-helpers')
+const { NotFound } = require('..')
+const AzureDevOpsBase = require('./azure-devops-base')
+const { keywords } = require('./azure-devops-helpers')
+
+const getPackagesSchema = Joi.object({
+  count: Joi.number().required(),
+  value: Joi.array()
+    .items(
+      Joi.object({
+        id: Joi.string().required(),
+        normalizedName: Joi.string().required(),
+        protocolType: Joi.string().required(),
```

Comment:
Given that we're reliant on the literal value of this field for various logic, and that it is presumable a finite set (I assume this maps to the handful of types AzDO supports), I think we should probably use a stricter schema here. For example:

```suggestion
        protocolType: Joi.allow('NuGet', '...').required(),
```

This will give us a louder notice should Microsoft make any changes to this